cli: TestDebugCheckStore failed

cli.TestDebugCheckStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/2d558644-0549-4278-9242-97961345075c?testReportRun=22&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [f117eea22dd7be380c7141cdf6cd7aba92dd9c70](https://github.com/cockroachdb/cockroach/commits/f117eea22dd7be380c7141cdf6cd7aba92dd9c70):


```
=== RUN   TestDebugCheckStore
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestDebugCheckStore1090274282
    test_log_scope.go:81: use -show-logs to present logs inline
    debug_check_store_test.go:90: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/cli/debug_check_store_test.go:90
        	Error:      	Received unexpected error:
        	            	check-store found problems
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.init
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/debug_check_store.go:55
        	            	  | runtime.doInit1
        	            	  | 	GOROOT/src/runtime/proc.go:7206
        	            	  | runtime.doInit
        	            	  | 	GOROOT/src/runtime/proc.go:7173
        	            	  | runtime.main
        	            	  | 	GOROOT/src/runtime/proc.go:253
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) check-store found problems
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestDebugCheckStore
        	Messages:   	dir=/var/lib/engflow/worker/work/2/exec/_tmp/10a3dedc16573dd48c5adba69877df01/TestDebugCheckStore1258554494/s1
        	            	out=r18:/Table/1{5-6} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=4]: stats inconsistency:
        	            	- stored:
        	            	{ContainsEstimates:0 LastUpdateNanos:1712989308216988618 LockAge:0 GCBytesAge:89851 LiveBytes:2897 LiveCount:45 KeyBytes:2601 KeyCount:55 ValBytes:2323 ValCount:90 IntentBytes:0 IntentCount:0 LockBytes:0 LockCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:1466 SysCount:8 AbortSpanBytes:68}
        	            	- recomputed:
        	            	{ContainsEstimates:0 LastUpdateNanos:1712989308216988618 LockAge:0 GCBytesAge:89851 LiveBytes:2897 LiveCount:45 KeyBytes:2601 KeyCount:55 ValBytes:2323 ValCount:90 IntentBytes:0 IntentCount:0 LockBytes:0 LockCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:1319 SysCount:8 AbortSpanBytes:68}
        	            	- diff:
        	            	SysBytes: 1466 != 1319
        	            	scanned 68 ranges (1 with estimates), total stats contains_estimates:0 last_update_nanos:1712989308687825374 lock_age:0 gc_bytes_age:1492250 live_bytes:1783393 live_count:9169 key_bytes:503798 key_count:9463 val_bytes:1314297 val_count:9854 intent_bytes:68 intent_count:1 lock_bytes:0 lock_count:1 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:37944 sys_count:416 abort_span_bytes:68 
        	            	
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestDebugCheckStore1090274282
--- FAIL: TestDebugCheckStore (61.52s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDebugCheckStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37803

cli.TestDebugCheckStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/568fb07f-eeb6-4b13-a13b-6e090babf5f3?testReportRun=8&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [d8073b58467ddb4c5b2ebe4a485fc42fb98f7200](https://github.com/cockroachdb/cockroach/commits/d8073b58467ddb4c5b2ebe4a485fc42fb98f7200):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestDebugCheckStore (14m57s)

```
Stack: 

```
goroutine 10955309 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestDebugCheckStore
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestDebugCheckStore3141802188
    test_log_scope.go:81: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123451 cli: TestDebugCheckStore failed [C-bug C-test-failure O-robot P-3 T-kv]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDebugCheckStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

cli.TestDebugCheckStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/2cad449a-e9d1-4527-82c5-50c70810fff3?testReportRun=11&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [3d9eaa7939e4fb5aa0518ba7e19750a55c93cf11](https://github.com/cockroachdb/cockroach/commits/3d9eaa7939e4fb5aa0518ba7e19750a55c93cf11):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestDebugCheckStore (14m57s)

```
Stack: 

```
goroutine 14373569 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestDebugCheckStore
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestDebugCheckStore3856725973
    test_log_scope.go:81: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=11</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123451 cli: TestDebugCheckStore failed [C-bug C-test-failure O-robot P-3 T-kv no-test-failure-activity]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDebugCheckStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/cli/BUILD.bazel
+++ b/pkg/cli/BUILD.bazel
@@ -358,7 +358,10 @@ go_test(
     embed = [":cli"],
     exec_properties = select({
         "//build/toolchains:is_heavy": {"test.Pool": "large"},
-        "//conditions:default": {"test.Pool": "default"},
+        # For the benefit of TestDebugCheckStore. That test is skipped under
+        # race/deadlock, so the "default" configuration is higher than the
+        # "is_heavy" configuration.
+        "//conditions:default": {"test.Pool": "heavy"},
     }),
     shard_count = 16,
     deps = [

